sdks/python/apache_beam/transforms/combiners_test.py

```diff
@@ -399,6 +432,108 @@ def test_global_fanout(self):
           | beam.CombineGlobally(combine.MeanCombineFn()).with_fanout(11))
       assert_that(result, equal_to([49.5]))
 
+  # Test that three different kinds of metrics work with a customized
+  # CounterIncrememtingCombineFn.
+  def test_simple_combine(self):
+    p = TestPipeline()
+    input = (p
+             | beam.Create([('c', 'b'),
+                            ('c', 'be'),
+                            ('c', 'bea'),
+                            ('d', 'beam'),
+                            ('d', 'apache')]))
+
+    # The result of concatenating all values regardless of key.
+    global_concat = (input
+                     | beam.Values()
+                     | beam.CombineGlobally(CounterIncrememtingCombineFn()))
+
+    # The (key, concatenated_string) pairs for all keys.
+    concat_per_key = (input | beam.CombinePerKey(
+        CounterIncrememtingCombineFn()))
+
+    result = p.run()
+    result.wait_until_finish()
+
+    # Verify the concatenated strings are correct.
+    expected_concat_per_key = [('c', 'bbebea'), ('d', 'beamapache')]
+    assert_that(global_concat, equal_to(['bbebeabeamapache']),
+                label='global concat')
+    assert_that(concat_per_key, equal_to(expected_concat_per_key),
+                label='concat per key')
+
+    # Verify the values of metrics are correct.
+    word_counter_filter = MetricsFilter().with_name('word_counter')
+    query_result = result.metrics().query(word_counter_filter)
+    if query_result['counters']:
+      word_counter = query_result['counters'][0]
+      self.assertEqual(word_counter.result, 5)
+
+    word_lengths_filter = MetricsFilter().with_name('word_lengths')
+    query_result = result.metrics().query(word_lengths_filter)
+    if query_result['counters']:
+      word_lengths = query_result['counters'][0]
+      self.assertEqual(word_lengths.result, 16)
+
+    word_len_dist_filter = MetricsFilter().with_name('word_len_dist')
+    query_result = result.metrics().query(word_len_dist_filter)
+    if query_result['distributions']:
+      word_len_dist = query_result['distributions'][0]
+      self.assertEqual(word_len_dist.result.mean, 3.2)
```

Comment:
To avoid spurious failures due to floating point issues, choose data such that the results can be represented exactly (e.g. as an integer or binary fraction like 3.5 or 3.25).

Reply:
Done.